crates/guest/src/guest.rs

```diff
@@ -80,9 +83,12 @@ where
 
     // Deserialize the host bytes into the output type.
     let bytes: Vec<u8> = crate::allocation::consume_bytes(output_guest_ptr);
-    match holochain_serialized_bytes::decode(&bytes) {
-        Ok(output) => Ok(output),
-        Err(_) => Err(WasmError::Deserialize(bytes)),
+    match holochain_serialized_bytes::decode::<Vec<u8>, Result<O, WasmError>>(&bytes) {
```

Comment:
this line is the main point of this PR, it changes the decode target from `O` to `Result<O, WasmError>` which is what core produces now that it returns results for all host functions